Makefile.PL

```diff
@@ -1,82 +1,264 @@
+# Dist::Zilla::Plugin::MakeMaker::Awesome 0.49.
+# Don't edit it but the dist.ini and plugins used to construct it.
+
 use strict;
 use warnings;
 
-use 5.010;
-
 use ExtUtils::MakeMaker;
 
+use FindBin '$Bin';
+use lib $Bin;
+
+$|++;
+my %requires;
+my %args;
+$args{LICENSE} = 'unrestricted';
+# clean generated test files
+$args{clean} = {FILES => "t/test-dat*"};
+
+# On Win32 (excluding cygwin) we know that IO::Socket::INET,
+# which is needed for keyserver stuff, doesn't work. In fact
+# it potentially hangs forever. So bail out with a N/A on
+# Win32.
+if ( $^O eq 'MSWin32' and 0 ) {
+       print "Keyserver behaviour is dangerous unreliable on Win32\n";
+       print "Not installing on this platform.\n";
+       exit(255);
+} else {
+       $requires{'IO::Socket::INET'} = 0;
+}
+
+# We will need something to handle SHA1/256
+unless (
+       can_use('Digest::SHA')  or
+       can_use('Digest::SHA::PurePerl') or
+       (can_use('Digest::SHA1') and can_use('Digest::SHA256'))
+) {
+       # Nothing installed, we need to install a digest module
+       if ( can_cc() ) {
+               $requires{'Digest::SHA'} = 0;
+       } else {
+               $requires{'Digest::SHA::PurePerl'} = 0;
+       }
+}
+
+# Is openpgp currently installed
+if ( can_use ('Crypt::OpenPGP') ) {
+       # Crypt::OpenPGP installed/available, continue on...
+} elsif ( my $gpg = locate_gpg() ) {
+       # We SHOULD have gpg, double-check formally
+       requires_external_bin ($gpg);
+} elsif ( can_cc() and $ENV{AUTOMATED_TESTING} ) {
+       # Dive headlong into a full Crypt::OpenPGP install.
+       $requires{'Crypt::OpenPGP'} = 0;
+} else {
+       # Ask the user what to do
+       ask_user();
+}
+
+unless ( can_run('diff') ) {
```

Comment:
Test::Diff seems to work fine according to the smokers so maybe require a modern version